File: vnes-emulator/src/main/java/vnes/emulator/channels/IAudioContext.java
package vnes.emulator.channels;

import vnes.emulator.MemoryMapper;

public interface IAudioContext {
    /**
     * Get the IRQ requester for interrupt handling.
     * @return The IRQ requester
     */
    IIrqRequester getIrqRequester();
    
    /**
     * Get the DMC sampler for sample loading operations.
     * @return The DMC sampler
     */
    DMCSampler getDmcSampler();

    int getSampleRate();
    void clockFrameCounter(int cycles);
    void updateChannelEnable(int value);
    
    // Method needed by channels to get length counter values
    int getLengthMax(int value);
}

File: vnes-emulator/src/main/kotlin/vnes/emulator/papu/IAudioContext.kt
package vnes.emulator.papu

interface IAudioContext {
    /**
     * Get the IRQ requester for interrupt handling.
     * @return The IRQ requester
     */
    val irqRequester: IIrqRequester

    /**
     * Get the DMC sampler for sample loading operations.
     * @return The DMC sampler
     */
    val dmcSampler: DMCSampler
    val sampleRate: Int
    fun clockFrameCounter(cycles: Int)
    fun updateChannelEnable(value: Int)

    // Method needed by channels to get length counter values
    fun getLengthMax(value: Int): Int
}